crypto/x509/x509_vfy.c

```diff
@@ -237,11 +237,27 @@ static int verify_chain(X509_STORE_CTX *ctx)
     return ok;
 }
 
+int X509_STORE_CTX_verify(X509_STORE_CTX *ctx)
+{
+    if (ctx == NULL) {
+        ERR_raise(ERR_LIB_X509, ERR_R_PASSED_NULL_PARAMETER);
+        ctx->error = X509_V_ERR_INVALID_CALL;
+        return -1;
+    }
+    if (ctx->cert == NULL && sk_X509_num(ctx->untrusted) >= 1)
+        ctx->cert = sk_X509_value(ctx->untrusted, 0);
```

Comment:
Don't you want to unmodify the context?

Reply:
The context cannot be reused for further verification anyway, so I do not see the point.

Reply:
> The context cannot be reused

Perhaps if you're checking for NULL then you might add an internal dirty/clean flag and use/check that.

Reply:
Good point! Done.

Reply:
Good point that the context might be reused!
Since `X509_STORE_CTX_cleanup()` does not affect the target `cert` field,
I've made sure the field is restored to NULL here.